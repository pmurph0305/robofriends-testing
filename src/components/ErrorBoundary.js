import React, { Component } from 'react';

class ErrorBoundary extends Component {
    constructor(props) {
        super(props);
        this.state = {
            hasError: false,
        }
    }

    componentDidCatch(error, info) {
        this.setState({ hasError: true });
    }

    render() {
        if (this.state.hasError) {
            return <h1>Error.</h1>
        }
        // Children will be the things between error boundary. Ie the cardlist.
        return this.props.children;
    }
}

export default ErrorBoundary;
