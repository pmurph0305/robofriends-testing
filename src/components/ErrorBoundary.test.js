import { shallow, mount } from 'enzyme';
import React from 'react';
import ErrorBoundary from './ErrorBoundary';

describe("tests for error boundary", () => {

    it('Render ErrorBoundary Component', () => {
        expect(shallow(<ErrorBoundary/>)).toMatchSnapshot();
    })

    it('Component catches correctly', () => {
        const mockError = new Error('test error');
        const ErrorComponent = () => null;
        const wrapper = mount(<ErrorBoundary><ErrorComponent/></ErrorBoundary>);
        wrapper.find(ErrorComponent).simulateError(mockError);
        expect(wrapper.state('hasError')).toEqual(true);
    })
})