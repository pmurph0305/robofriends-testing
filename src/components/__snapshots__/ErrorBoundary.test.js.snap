// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for error boundary Render ErrorBoundary Component 1`] = `ShallowWrapper {}`;
